pkg/ddc/base/interface.go

```diff
@@ -0,0 +1,32 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package base
+
+import (
+	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+type RuntimeInterface interface {
```

Comment:
I am a little confused about the differences between `RuntimeInterface` and `RuntimeInfoInterface`. So which info should be included in the `RuntimeInterface` and which should be included in the other? Can we refactor these two into one struct?

Reply:
In the end, I hope every thing will go to `RuntimeInterface`. But for now, the key differences between them is :

1. `RuntimeInterface` is instance of `runtime.Object`, so it can be used by client-go to update in API Server. All the runtime can extend it.
2. `Info` for now just be information of the runtime which engine is uses it for handle.  Once we move the key logic to ctrl package, we can start to deprecate `RuntimeInfoInterface`.

Reply:
Thx. LGTM